synse/http.go

```diff
@@ -31,6 +30,20 @@ func NewHTTPClient(options *Options) (Client, error) {
 	}, nil
 }
 
+// NewHTTPClientV3 returns a new instance of a http client with API v3.
+func NewHTTPClientV3(options *Options) (Client, error) {
+	client, err := createHTTPClient(options)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create a http client")
+	}
+
+	return &httpClient{
+		options:    options,
+		client:     client,
+		apiVersion: "v3",
+	}, nil
+}
```

Comment:
To reduce duplication of code (especially if/when we have multiple constructors), we could call the default constructor to create the client, then just update the version prior to returning:
```go
func NewHTTPClientV3(options *Options) (Client, error) {
	client, err := NewHTTPClient(options)
	if err != nil {
		return nil, err
	}
	client.apiVersion = "v3"
	return client, nil
}
```

Reply:
`NewHTTPClient` returns an `httpClient`, which implements the `Client` interface. Since `apiVersion` is an unexported field of `httpClient` only, setting its value prior to its returning is not allowed.

So maybe what I could do instead is to define `SetAPIVersion()` and `GetAPIVersion()` for `Client` interface that all client should implement to set/get their internal `apiVersion` value, if that makes sense.

Reply:
If we have the get/set api version as I mentioned above, another use case could be something like:
```go
// Create a new HTTP Client that use Synse API v3.
client := NewHTTPClientV3(options)

// Do something with the client...

// Have the same client use a different api version, for example v2 for some reasons.
client.SetAPIVersion("v2")

// Do something with the client...
```
It still makes sense in this case, though maybe it's unlikely to happen in real world. So the question is: Do we want to expose these get/set methods or using can just them internally is fine enough?

Reply:
Oh right, I forgot that `Client` is the interface. You can ignore this suggestion. Sorry!